### ISSUE TITLE
Build Brigade container images for PRs

### ISSUE THREAD

**Issue by radu-matei** [brigadecore/brigade#761]
When testing PRs, it's really helpful to have built container images for Brigade components - meaning as a reviewer I don't have to build and push the image myself, plus we can easily rollback to any previous PR.

We could keep the intermediate images at least during a release cycle, and when we cut a new one, simply delete all PR images from the previous release cycle?

What are the downsides for this?
cc @vdice

**Comment by dgkanatsios** [comment]
Great idea, I think that there would also be a need for the respective Helm templates to be created as well for easier testing, right?

**Comment by radu-matei** [comment]
> Ideally, this would spark a refresh/revive of this project's functional/e2e test approach, which we could also add in to the PR checks pipeline, to run after component image publishing.

This would be fantastic!

**Comment by krancour** [comment]
This should all be substantially easier now that a bunch of pipeline improvements went in a couple months ago. In the current state, we publish images tagged with git sha and the mutable tag "edge" on every merge to master. All we need to do is change things around so that publishing of sha-tagged images occurs toward the end of the PR pipeline (after unit tests, before e2e tests). If we consider the moving target "edge" to be representative of what's at the head of master, that should still only be published upon merge to master.

Long story short... this should be pretty easy to accomplish and I'd be happy to tackle this.

One thing I would suggest, however, is using a separate repository (image name) for images built from PRs to very unambiguously convey that the images were built from code that may be in a completely unvetted state.

**Comment by krancour** [comment]
Closing due to staleness. Please do re-open if appropriate.
